src/Neo/Cryptography/Base58.cs

```diff
@@ -100,20 +119,35 @@ public static string Encode(ReadOnlySpan<byte> input)
             BigInteger value = new(input, isUnsigned: true, isBigEndian: true);
 
             // Encode BigInteger to Base58 string
-            var sb = new StringBuilder();
+            var sb = new StringBuilder(input.Length * 138 / 100 + 5);
 
             while (value > 0)
             {
-                value = BigInteger.DivRem(value, Alphabet.Length, out var remainder);
-                sb.Insert(0, Alphabet[(int)remainder]);
+                value = BigInteger.DivRem(value, s_alphabetLength, out var remainder);
+                sb.Append(Alphabet[(int)remainder]);
             }
 
             // Append `1` for each leading 0 byte
             for (int i = 0; i < input.Length && input[i] == 0; i++)
             {
-                sb.Insert(0, Alphabet[0]);
+                sb.Append(ZeroChar);
             }
-            return sb.ToString();
+
+            Span<char> copy = stackalloc char[sb.Length];
+            sb.CopyTo(0, copy, sb.Length);
+            copy.Reverse();
```

Comment:
is not better to do the `Insert(0,..` and avoid this revert?

Reply:
No, it is not. Insertion to 0 copies whole buffer to +1.

Reply:
Also, it's proven by my benchmarks that appending then reversing is faster